README.md

```diff
@@ -79,142 +54,165 @@ $ open QuickStart.xcworkspace
+                               [Users addObject:s];
+                           }
+                           
+                           completionBlock(Users, nil);
+                       }
+                       else
+                       {
+                           completionBlock(nil, error);
+                       }
+                       
+                   }];
+}
+```
 
-    NSString *graphURL = [NSString stringWithFormat:@"%@%@/users?api-version=%@&$filter=startswith(userPrincipalName, '%@')", data.taskWebApiUrlString, data.tenant, data.apiversion, searchString];
-
-
-    [self craftRequest:[self.class trimString:graphURL]
-                parent:parent
-     completionHandler:^(NSMutableURLRequest *request, NSError *error) {
-
-         if (error != nil)
-         {
-             completionBlock(nil, error);
-         }
-         else
-         {
-
-             NSOperationQueue *queue = [[NSOperationQueue alloc]init];
-
-             [NSURLConnection sendAsynchronousRequest:request queue:queue completionHandler:^(NSURLResponse *response, NSData *data, NSError *error) {
-
-                 if (error == nil && data != nil){
-
-                     NSDictionary *dataReturned = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
-
-                     // We can grab the top most JSON node to get our graph data.
-                     NSArray *graphDataArray = [dataReturned objectForKey:@"value"];
-
-                     // Don't be thrown off by the key name being "value". It really is the name of the
-                     // first node. :-)
-
-                     //each object is a key value pair
-                     NSDictionary *keyValuePairs;
-                     NSMutableArray* Users = [[NSMutableArray alloc]init];
-
-                     for(int i =0; i < graphDataArray.count; i++)
-                     {
-                         keyValuePairs = [graphDataArray objectAtIndex:i];
 
-                         User *s = [[User alloc]init];
-                         s.upn = [keyValuePairs valueForKey:@"userPrincipalName"];
-                         s.name =[keyValuePairs valueForKey:@"givenName"];
+4. Build and Run the application
 
-                         [Users addObject:s];
-                     }
 
-                     completionBlock(Users, nil);
-                 }
-                 else
-                 {
-                     completionBlock(nil, error);
-                 }
 
-             }];
-         }
-     }];
+Congratulations! You now have a working iOS application using a third party library that has the ability to authenticate users, securely call Web APIs using OAuth 2.0, and get basic information about the user.  If you haven't already, now is the time to populate your tenant with some users.  Run your QuickStart app, and sign in with one of those users.  Search for other users based on their UPN.  Close the app, and re-run it.  Notice how the user's session remains intact.
```

Comment:
Earlier this seemed to be a directory searcher app, but this alludes to it just calling the /me endpoint. Can you clarify what the app does?